### PR TITLE
Add `MiniscopeImagingInterface`

### PR DESCRIPTION
**Pull request by weiglszonja**
Adds `MiniscopeImagingInterface` for Miniscope imaging data stream. 

- Uses [ndx-miniscope](https://github.com/catalystneuro/ndx-miniscope) to add a Miniscope device from the device metadata and the imaging data as `OnePhotonSeries`
- More info about the imaging data stream is at https://github.com/catalystneuro/ndx-miniscope

- [x] Remove ndx-miniscope pin after https://github.com/catalystneuro/ndx-miniscope/pull/11 is merged

### CODE REVIEW

File: tests/test_on_data/test_imaging_interfaces.py [hunk 1]
```diff
@@ -228,3 +232,50 @@ def check_read_nwb(self, nwbfile_path: str):
...
+            self.assertIsInstance(device, Miniscope)
+            self.assertEqual(device.fields, self.device_metadata)
+            self.assertEqual(device, nwbfile.imaging_planes[self.imaging_plane_name].device)
+
+            # Check OnePhotonSeries
+            self.assertIn(self.photon_series_name, nwbfile.acquisition)
+            self.assertEqual(nwbfile.acquisition[self.photon_series_name].unit, "px")
+
+        imaging = self.interface.imaging_extractor
+        nwb_imaging = NwbImagingExtractor(file_path=nwbfile_path, optical_series_name="OnePhotonSeries")
```
Comment (weiglszonja) [inline tests/test_on_data/test_imaging_interfaces.py:279]: this also requires a refactor on the `roiextractors` side https://github.com/catalystneuro/roiextractors/pull/229

**Comment by weiglszonja** [comment]
The test data on gin has to be fixed because it includes the timestamps for all the frames while the .avi files only have the first 5 frames. I tried to fix it in branch, but the changes are not recognized: https://gin.g-node.org/CatalystNeuro/ophys_testing_data/compare/fix_miniscope...CatalystNeuro:main

**Comment by weiglszonja** [comment]
> The test data on gin has to be fixed because it includes the timestamps for all the frames while the .avi files only have the first 5 frames. I tried to fix it in branch, but the changes are not recognized: https://gin.g-node.org/CatalystNeuro/ophys_testing_data/compare/fix_miniscope...CatalystNeuro:main

This was fixed and is tested in https://github.com/catalystneuro/neuroconv/pull/468/commits/618c5d9f9d55d32a13022fa172410762678f9ffd

**Comment by CodyCBakerPhD** [comment]
@weiglszonja Ready for final review?

File: src/neuroconv/datainterfaces/ophys/miniscope/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ndx-miniscope @ git+https://github.com/catalystneuro/ndx-miniscope.git@7de541ebceca7e73571ad7aecafb3ec6b8a19c16#egg=ndx-miniscope
```
Comment (weiglszonja) [inline src/neuroconv/datainterfaces/ophys/miniscope/requirements.txt:1]:
```suggestion
ndx-miniscope>=0.5.1
```

File: src/neuroconv/datainterfaces/ophys/requirements.txt [hunk 3]
```diff
@@ -1 +1 @@
-roiextractors>=0.5.2
+roiextractors @ git+https://github.com/catalystneuro/roiextractors.git@51ccccbb53fbab7e389188b60693b0e5633746bf#egg=roiextractors
```
Comment (CodyCBakerPhD) [inline src/neuroconv/datainterfaces/ophys/requirements.txt:1]:
```suggestion
roiextractors>=0.5.3
```

Went ahead and cut a new release for this

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
@weiglszonja Really good work on this!

**Pull request merged**